pkg/docker/docker.go

```diff
@@ -111,6 +131,7 @@ type PullResult struct {
 
 // RunContainerOptions are options passed in to the RunContainer method
 type RunContainerOptions struct {
+	Name             string
```

Comment:
Looked at how we'd leverage this from Origin, doesn't seem useful.

We never set `RunContainerOptions` directly, instead we call the `Save` and `Execute` methods defined in `pkg/build/strategies/sti/sti.go`, and they call `RunContainer` with options based on yet another `Config` struct.

If we don't care to correlate BuildConfigs and S2I containers, then the name generated by `containerName` should be enough. Otherwise we'd need to make changes to the `Config` struct. In that case, probably instead of providing a container name we should provide just metadata because S2I might create multiple containers and I guess we don't want to use a single name (for a given build).